components/script/dom/bluetoothremotegattcharacteristic.rs

```diff
@@ -175,27 +169,26 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
     fn ReadValue(&self) -> Fallible<ByteString> {
         let (sender, receiver) = ipc::channel().unwrap();
         if !self.Service().Device().Gatt().Connected() {
-            Err(Network)
-        } else {
-            self.get_bluetooth_thread().send(
-                BluetoothMethodMsg::ReadValue(self.get_instance_id(), sender)).unwrap();
-            let result = receiver.recv().unwrap();
-            let value = match result {
-                BluetoothObjectMsg::BluetoothReadValue {
-                    value
-                } => {
-                    Some(ByteString::new(value))
-                },
-                BluetoothObjectMsg::Error {
-                    error
-                } => {
-                    return Err(Type(error))
-                },
-                _ => unreachable!()
-            };
-            *self.value.borrow_mut() = value;
-            Ok(self.GetValue().unwrap())
+            return Err(Network)
         }
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::ReadValue(self.get_instance_id(), sender)).unwrap();
+        let result = receiver.recv().unwrap();
+        let value = match result {
+            BluetoothObjectMsg::BluetoothReadValue {
+                value
+            } => {
+                Some(ByteString::new(value))
```

Comment:
Maybe we can leave the Some here so we dont need the unwrap() at the end of the function.
```
        let value = match result {
            BluetoothObjectMsg::BluetoothReadValue {
                value
            } => {
                ByteString::new(value)
            },
            BluetoothObjectMsg::Error {
                error
            } => {
                return Err(Type(error))
            },
            _ => unreachable!()
        };
        *self.value.borrow_mut() = Some(value.clone());
        Ok(value)
    }
```